react-hash-routing/client/app.jsx

```diff
@@ -15,12 +15,14 @@ export default function App() {
      * it with the parseRoute() function and update state
      * Clean up when the component unmounts.
      */
-    window.addEventListener('hashchange', () => {
+    function hash() {
       const route = parseRoute(window.location.hash);
       setRoute(route);
-    });
-    return () => window.removeEventListener('hashchange');
-  }, []);
+    }
+
+    window.addEventListener('hashchange', hash);
+    return () => window.removeEventListener('hashchange', hash);
+  }, [route]);
```

Comment:
This dependency causes the event listener to be added and removed every time the route changes. This is OK, but it is a bit excessive. Your code in `useEffect` does not depend on `route` so it doesn't need to be a dependency (the `route` variable on line 19 is not the same as the `route` variable defined in line 9 and referenced in line 25, they just happen to have the same name).

Typically for something like this, you'd just pass `[]` as your dependency array.